lib/spack/spack/compiler.py

```diff
@@ -454,36 +466,46 @@ def required_libs(self):
+            self._compile_c_source_output = self._compile_stub_c_source()
         return self._compile_c_source_output
 
-    def _compile_dummy_c_source(self) -> Optional[str]:
-        cc = self.cc if self.cc else self.cxx
-        if not cc or not self.verbose_flag:
-            return None
-
-        try:
-            tmpdir = tempfile.mkdtemp(prefix="spack-implicit-link-info")
-            fout = os.path.join(tmpdir, "output")
-            fin = os.path.join(tmpdir, "main.c")
-
-            with open(fin, "w") as csource:
-                csource.write(
-                    "int main(int argc, char* argv[]) { (void)argc; (void)argv; return 0; }\n"
-                )
-            cc_exe = spack.util.executable.Executable(cc)
-            for flag_type in ["cflags" if cc == self.cc else "cxxflags", "cppflags", "ldflags"]:
-                cc_exe.add_default_arg(*self.flags.get(flag_type, []))
-
-            with self.compiler_environment():
-                return cc_exe(self.verbose_flag, fin, "-o", fout, output=str, error=str)
-        except spack.util.executable.ProcessError as pe:
-            tty.debug("ProcessError: Command exited with non-zero status: " + pe.long_message)
+    def _compile_stub_c_source(self) -> Optional[str]:
+        lang = "cc" if self.cc else "cxx"
+        if not self.verbose_flag or not (self.cc or self.cxx):
             return None
-        finally:
-            shutil.rmtree(tmpdir, ignore_errors=True)
+        flags = [self.verbose_flag]
+        for flag_type in ["cflags" if lang == "cc" else "cxxflags", "cppflags", "ldflags"]:
+            flags.extend(self.flags.get(flag_type, []))
+        cmp_args = ["-o", "output"]
+        return self.try_compiler(lang, cmp_args=cmp_args, flags=flags)
+
+    # def _compile_dummy_c_source(self) -> Optional[str]:
+    #     cc = self.cc if self.cc else self.cxx
+    #     if not cc or not self.verbose_flag:
+    #         return None
+
+    #     try:
+    #         tmpdir = tempfile.mkdtemp(prefix="spack-implicit-link-info")
+    #         fout = os.path.join(tmpdir, "output")
+    #         fin = os.path.join(tmpdir, "main.c")
+
+    #         with open(fin, "w") as csource:
+    #             csource.write(
+    #                 "int main(int argc, char* argv[]) { (void)argc; (void)argv; return 0; }\n"
+    #             )
+    #         cc_exe = spack.util.executable.Executable(cc)
+    #         for flag_type in ["cflags" if cc == self.cc else "cxxflags", "cppflags", "ldflags"]:
+    #             cc_exe.add_default_arg(*self.flags.get(flag_type, []))
+
+    #         with self.compiler_environment():
+    #             return cc_exe(self.verbose_flag, fin, "-o", fout, output=str, error=str)
+    #     except spack.util.executable.ProcessError as pe:
+    #         tty.debug("ProcessError: Command exited with non-zero status: " + pe.long_message)
+    #         return None
+    #     finally:
+    #         shutil.rmtree(tmpdir, ignore_errors=True)
```

Comment:
Yes, dead code as I wanted to refer back to this in case there was an issue with the `asp.py` changes. Like I mention in another comment, this PR wasn't ready to go in, I was in the middle of an iteration when work was paused, so I left a lot of cruft in in case I needed to refer to this.